### PR TITLE
Migrate to spine 0.10.54

### PR DESCRIPTION
**Pull request by mdrachuk**
This PR updates the examples to the latest Spine and its dependencies.

What was done:

1. `core` dependency updated to 0.10.54-SNAPSHOT, and `base` updated to 0.10.51-SHAPSHOT.
2. Gradle version updated to 4.9.
3. `config` submodule is added.
4. Migrated to JUnit 5.
5. @CheckReturnValue is added to all package-info files.
6. `final` is removed where it is redundant.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Fri Aug 12 21:04:22 EEST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.8.1-bin.zip
```
Comment (armiol) [inline gradle/wrapper/gradle-wrapper.properties]: Have you modified this file by hand? The absence of the first line with the timestamp comment is suspicious. 

If so, please use the procedure (see Gradle documentation on updating the wrapper) instead.

File: hello-world/src/main/java/io/spine/helloworld/package-info.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+@CheckReturnValue
+@ParametersAreNonnullByDefault
+package io.spine.helloworld;
+
+import com.google.errorprone.annotations.CheckReturnValue;
+
+import javax.annotation.ParametersAreNonnullByDefault;
```
Comment (armiol) [inline hello-world/src/main/java/io/spine/helloworld/package-info.java]: Please have a comment describing the package.

File: server-examples/src/main/java/io/spine/examples/aggregate/client/package-info.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+@CheckReturnValue
+@ParametersAreNonnullByDefault
+package io.spine.examples.aggregate.client;
+
+import com.google.errorprone.annotations.CheckReturnValue;
+
+import javax.annotation.ParametersAreNonnullByDefault;
```
Comment (armiol) [inline server-examples/src/main/java/io/spine/examples/aggregate/client/package-info.java]: Same here. See how we document the packages in `core-java` or `base` repositories.

File: server-examples/src/main/java/io/spine/examples/tuple/KanbanBoardAggregate.java [hunk 4]
```diff
@@ -101,21 +109,24 @@ private TaskAdded taskAdded(BoardId id, TaskId taskId) {
     }
 
     @Apply
+    @SuppressWarnings("CheckReturnValue")
```
Comment (armiol) [inline server-examples/src/main/java/io/spine/examples/tuple/KanbanBoardAggregate.java]: This should not be a problem anymore. See https://github.com/SpineEventEngine/base/pull/138.

Please remove the annotation from the appliers. There is also an IDEA inspection that detects redundant warning suppressions, so you can use that.

File: server-examples/src/test/java/io/spine/examples/aggregate/server/ServerTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static io.spine.core.BoundedContextNames.newName;
+import static io.spine.examples.storage.gae.LocalDatastoreEnvironment.buildDatastore;
+
+@SuppressWarnings({"InnerClassMayBeStatic", "ClassCanBeStatic" /* JUnit nested classes cannot be static. */})
+@DisplayName("Server should")
+class ServerTest {
```
Comment (armiol) [inline server-examples/src/test/java/io/spine/examples/aggregate/server/ServerTest.java]: Please find out who's the `@author` (there may be several engineers) and fill in the tag value.

It concerns all the source files.

File: server-examples/src/test/java/io/spine/examples/aggregate/server/ServerTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        @DisplayName("Google App Engine storage")
+        void gae() throws Exception {
+            Datastore localDatastore = buildDatastore();
+            Supplier<StorageFactory> supplier = StorageFactorySuppliers.googleDatastore(localDatastore, false);
+            Server server = Server.newInstance(supplier, false);
+    
+            runServer(server);
+        }
+    }
+
```
Comment (armiol) [inline server-examples/src/test/java/io/spine/examples/aggregate/server/ServerTest.java]: Please kill this redundant line.

File: server-examples/src/test/java/io/spine/examples/tuple/ColumnTest.java [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+        @Test
+        @SuppressWarnings("CheckReturnValue")
+        @DisplayName("IllegalArgumentException for unknown column")
+        void ofUnknownColumn() {
+            assertThrows(IllegalArgumentException.class,
+                         () -> Column.of(KanbanBoard.Column.UNKNOWN));
+        }
+
+        @Test
+        @SuppressWarnings("CheckReturnValue")
```
Comment (armiol) [inline server-examples/src/test/java/io/spine/examples/tuple/ColumnTest.java]: Same here.

**Comment by mdrachuk** [comment]
@armiol I’ve taken care of all the comments. PTAL.

**Pull request merged**